fenix/app/src/main/res/navigation/nav_graph.xml

```diff
@@ -29,6 +29,18 @@
         app:popUpTo="@id/homeFragment"
         app:popUpToInclusive="false" />
 
+    <action
+        android:id="@+id/action_global_home_juno_onboarding"
+        app:destination="@id/junoOnboardingFragment"
+        app:popUpTo="@id/startupFragment"
+        app:popUpToInclusive="true" />
+
+    <action
+        android:id="@+id/action_onboarding_home"
+        app:destination="@id/homeFragment"
+        app:popUpTo="@id/junoOnboardingFragment"
+        app:popUpToInclusive="true" />
```

Comment:
@rahulsainani @rocketsroger folks, there are some oddities here that I ran into when trying to debug a completely different issue which crossed into this Juno work.

* `action_global_home_juno_onboarding` this naming is confusing - if you look at the examples around this file, we name the action to where it is going to avoid misinterpretting the direction as "home juno onboarding" which could be HomeActivity or HomeFragment. So a better name would be `action_global_juno_onboarding`.
* `action_onboarding_home` is also confusing for the same reasons above. It reads (via the JunoOnboardingFragment code) that we are at the onboarding fragment and now going to the onboarding home direction. From what I understand, it looks like we want to go to the HomeFragment and remove the onboarding fragment from our backstack. Further, it is missing the "global" namespace which is what helps read the generated code better in the kotlin side.
* For both these actions, why do they need to be _global_ actions? 
  * `action_global_home_juno_onboarding` can be a action under the `junoOnboardingFragment` fragment declaration. No other fragment needs to be able to navigate to it, if I understand correctly (please correct me if I'm wrong).
  * `action_onboarding_home` might be unnecessary, I think? We have a global direction to go home right above called `action_global_home`, so maybe that is enough. I'm not certain though if that will do the same popUpTo action as intended by this action. If not, please do the same as above and remove it from the global actions to be a fragment direction.

Let me know if you'd like me to file a bug for the above mentioned as well. There are examples in this file for some of the recommendations.

Please take time to discuss and understand the implications of a change before merging it. Pushing back deadlines is less work that tech debt that can lead to bugs.

Reply:
I've created this [bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1830083) here to track and noted down my initial findings.